java/arcs/core/storage/keys/JoinStorageKey.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage.keys
+
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageKeyParser
+import arcs.core.storage.referencemode.embed
+import arcs.core.storage.referencemode.unEmbed
+
+/** Protocol to be used when the StorageKey is composed of multiple StorageKeys. */
+const val COMPOSITE_PROTOCOL = "join"
+
+class JoinStorageKey(
+    private val n: Int,
+    private val storageKeys: List<StorageKey>
+) : StorageKey(COMPOSITE_PROTOCOL) {
+    override fun toKeyString(): String {
+        val builder = StringBuilder()
+        builder.append("$n/")
+        storageKeys.forEach { storageKeys ->
```

Comment:
the variable in the lamda should be singular I think:
`storageKeys.forEach { storageKey ->`

I would even consider:
`storageKeys.forEach { builder.append("{${it.embed()}}") }`

Reply:
done.